vllm/model_executor/model_loader/utils.py

```diff
@@ -37,6 +51,34 @@ def get_model_architecture(
             and "MixtralForCausalLM" in architectures):
         architectures = ["QuantMixtralForCausalLM"]
 
+    vllm_supported_archs = ModelRegistry.get_supported_archs()
+    for i, arch in enumerate(architectures):
+        if arch == "TransformersModel":
+            continue
+        custom_module = None
+        auto_map = getattr(model_config.hf_config, "auto_map", None)
+        if auto_map is not None and hasattr(auto_map, "AutoModel"):
+            custom_module = get_class_from_dynamic_module(
+                model_config.hf_config.auto_map["AutoModel"],
+                model_config.model)
+        if model_config.model_impl == ModelImpl.TRANSFORMERS:
+            if not is_transformers_impl_compatible(arch, custom_module):
+                raise ValueError(
+                    "The Transformers implementation of %s is not compatible "
+                    "with vLLM.", arch)
+            architectures[i] = "TransformersModel"
+        if (model_config.model_impl == ModelImpl.AUTO
+                and arch not in vllm_supported_archs):
+            if not is_transformers_impl_compatible(arch, custom_module):
+                raise ValueError(
+                    "%s has no vLLM implementation and the Transformers "
+                    "implementation is not compatible with vLLM.", arch)
```

Comment:
The %s style doesn't work here like the logger
```suggestion
        if model_config.model_impl == ModelImpl.TRANSFORMERS:
            if not is_transformers_impl_compatible(arch, custom_module):
                raise ValueError(
                    f"The Transformers implementation of {arch} is not "
                    "compatible with vLLM.")
            architectures[i] = "TransformersModel"
        if (model_config.model_impl == ModelImpl.AUTO
                and arch not in vllm_supported_archs):
            if not is_transformers_impl_compatible(arch, custom_module):
                raise ValueError(
                    f"{arch} has no vLLM implementation and the Transformers "
                    "implementation is not compatible with vLLM.")
```